### PR TITLE
Group task - Item Class & Others

### LINKED ISSUE #2: [1pt] Add class properties (group task)
- All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._

### LINKED ISSUE #3: [0.5pt] Create main.rb file (group task)
- Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() method is false
 - Create a main.rb file that will serve as your console app entry-point.
 - Implement startup actions:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

### LINKED ISSUE #4: [1pt] class properties visible in the diagram (group task)
All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._

### PR DESCRIPTION
**Pull request by mmatongo**
### Group task
- Create Item class in a separate .rb file.
- All Item class properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- Add all methods visible in the diagram.
- Implement methods:
    - can_be_archived?() in the Item class
        - should return true if published_date is older than 10 years
        - otherwise, it should return false
    - move_to_archive() in the Item class
        - should reuse can_be_archived?() method
        - should change the archived property to true if the result of the can_be_archived?() method is true
        - should do nothing if the result of the can_be_archived?() method is false
 - Create a main.rb file that will serve as your console app entry-point.
 - Implement startup actions:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.

Closes #2
Closes #3 
Closes #4

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    puts ''
+
+    loop do
+      @options.each { |key, value| puts "#{key}) #{value}" }
+
+      option = gets.chomp
+      break if option == '13'
+    end
+  end
+end
```
Comment (julie-ify) [inline main.rb:31]: Hi Team, What do you think of implementing a prompt message where you prompt the user to select any number from the menu. This is a good way to improve our app and make it user friendly 😄 

Ours:
Code_ogGHgbKFTk

My suggestion:

Code_FNkklqPoaT

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class App
+  def initialize
+    @options = {
+      '1' => 'List all books',
+      '2' => 'List all music albums',
+      '3' => 'List all movie',
```
Comment (julie-ify) [inline main.rb:6]: There is a typo error here, should be `'3' => 'List all movies'`  😺

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi Team👋, 

Good job so far!

No linter errors ✔️ 
Github flow ✔️ 
Descriptive pull request message ✔️ 
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 2 above)
Comment (mmatongo) [reply main.rb]: done

File: main.rb (hunk 1 above)
Comment (mmatongo) [reply main.rb]: done

**Review by julie-ify — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**